### ISSUE TITLE
Unable to modify source for debugging

### ISSUE THREAD

**Issue by lucs** [perl6/DBIish#72]
I'm getting an error, and I'd like to insert 'print' statements in the source to try to debug, but I don't understand which source to modify and have run.

This is what I get with the panda installed DBIish:

    Month out of range. Is: 0, should be in 1..12
      in block  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/B874A114F16EBFE5EA4815E5E20CA48A7810B32C (DBDish::mysql::StatementHandle) line 178
      in code  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/B874A114F16EBFE5EA4815E5E20CA48A7810B32C (DBDish::mysql::StatementHandle) line 168
      in block  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/B874A114F16EBFE5EA4815E5E20CA48A7810B32C (DBDish::mysql::StatementHandle) line 165
      in method _row at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/B874A114F16EBFE5EA4815E5E20CA48A7810B32C (DBDish::mysql::StatementHandle) line 161
      in method fetchrow at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C790FDD5119FC8D71D36CB0BA92A32B7A0A6A77A (DBDish::StatementHandle) line 119
      in block  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C790FDD5119FC8D71D36CB0BA92A32B7A0A6A77A (DBDish::StatementHandle) line 159
      in method fetchall_arrayref at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C790FDD5119FC8D71D36CB0BA92A32B7A0A6A77A (DBDish::StatementHandle) line 169
      in block <unit> at -e line 15

I opened the 'B874A11…' file, and typed some junk on line 178 ('sdfasdf'), then ran the program again, but I got the exact same result.

Then I cloned the DBIish repo, and ran my program with -I/path/to/the/cloned/repo/lib, and this is what I get:

    ===SORRY!=== Error while compiling /opt/prj/c/cirano/ciw/repo/dev/site#sources/B874A114F16EBFE5EA4815E5E20CA48A7810B32C (DBDish::mysql::StatementHandle)
    Undeclared routine:
        sdfasdf used at line 178

      in block  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F (DBIish) line 41
      in any  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/precomp/C3D71A4CAC2ECF2C70B50A1F4436228B32794CA7.1476211372.19594/C7/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F line 1
      in any  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/precomp/C3D71A4CAC2ECF2C70B50A1F4436228B32794CA7.1476211372.19594/C7/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F line 1
      in block  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F (DBIish) line 54
      in any  at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/precomp/C3D71A4CAC2ECF2C70B50A1F4436228B32794CA7.1476211372.19594/C7/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F line 1
      in method install-driver at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F (DBIish) line 58
      in method connect at /opt/prj/l/p6/perl6-rakudo.git/install/share/perl6/site/sources/C7DE14D64CD73BC06BF8D3D436441E1B67A9F91F (DBIish) line 44
      in block <unit> at -e line 3

"/opt/prj/c/cirano/ciw/repo/dev" is the working directory from which I'm running the program.

Which sources should receive my eventual 'print' debugging statements, and how do I run my program given precompilation and all that?

I built my Perl 6 from scratch yesterday:
[perl6ver.txt](https://github.com/perl6/DBIish/files/524959/perl6ver.txt)

**Comment by niner** [comment]
Please do not modify files in your install/share/perl6 directory in any way. They are not meant to be modified by a user.

The correct way is to clone the DBIish repo, modify that copy and use it via -Ilib. If it does not pick up those files, the most probable explanation is that you mis-typed the include path and it fell back to the installed DBIish version and then stumbled over your previous modifications.

**Comment by lucs** [comment]
Turns out I did actually mis-type the include path: once corrected, things worked as expected. I didn't think of double checking the path because when I removed the -I/wrong/path/to/lib from my invocation to try it again, the error produced was the initial one, so I thought that somehow the path shown in the -I... was involved. Note that I was unlucky in using a wrong path that was correct up to the last components: -I/an/existing/subdir/DBDish/lib, instead of .../DBIish/lib. If I had used something like /no/such/dir/DBDish/lib, the error message would have made more sense to me: "Failed to create directory /no/such/dir/DBDish/lib/.precomp/...".

Thanks Stefan!
